### PR TITLE
ci: Add tests against the beta version of Flutter/Dart

### PR DESCRIPTION
**Pull request by Nidal-Bakir**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description

Closes: #888

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] A changelog entry

### CODE REVIEW

**Comment by Nidal-Bakir** [comment]
The beta falls because the upper bound

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -92,12 +95,22 @@ jobs:
...
             sdk: 3.3.x
+          - name: Flutter Beta
+            os: ubuntu-latest
+            sdk: beta
       fail-fast: false
     name: Test ${{ matrix.name }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+      - name: Setup Beta flutter
```
Comment (mtrezza) [inline .github/workflows/ci.yml:106]:
```suggestion
      - name: Setup flutter (beta)
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -92,12 +95,22 @@ jobs:
...
+        if: ${{  matrix.sdk == 'beta' }}
```
Comment (mtrezza) [inline .github/workflows/ci.yml:107]:
```suggestion
        if: ${{ matrix.sdk == 'beta' }}
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -92,12 +95,22 @@ jobs:
           - name: Flutter 3.3
             os: ubuntu-latest
             sdk: 3.3.x
+          - name: Flutter Beta
```
Comment (mtrezza) [inline .github/workflows/ci.yml:98]:
```suggestion
          - name: Flutter beta
```

**Review by mtrezza — APPROVED** [review]
Looks good! Let's see if the CI passes now. Otherwise we may need to fix

**Comment by Nidal-Bakir** [comment]
> Because parse_server_sdk_flutter depends on parse_server_sdk >=5.0.0 which requires SDK version >=2.18.0 <3.1.0, version solving failed.

The parse server Dart SDK has not been published yet.

**Comment by mtrezza** [comment]
Yes, so once https://github.com/parse-community/Parse-SDK-Flutter/issues/905 is merged and 5.1.1 is published, we can merge this.

**Pull request merged**